### DISCUSSION TITLE
Errors with shadow-dom branch

### DISCUSSION THREAD

**Discussion by maenkhouseh** [mubaidr/vite-vue3-chrome-extension-v3#150] [Ideas]
Once I run pnpm run dev after switching to the shadow-dom branch I get so many errors in console (and the extension losses style), the errors look as such:

```
transforming...
[0] 5:02:41 PM [vite] (client) Pre-transform error: [sass] File URL path must be absolute
[0]   ╷
[0] 2 │ @use '/node_modules/vite-plugin-vue-devtools/src/overlay/devtools-overlay.css';
[0]   │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[0]   ╵
[0]   src\assets\base.scss 2:1           @use
[0]   src\content-script\index.scss 1:1  root stylesheet
[0]   Plugin: vite:css
[0]   File: D:/dev-new/tendersalerts-extension/src/content-script/index.scss?inline
[0]   [sass] File URL path must be absolute
[0]     ╷
[0]   2 │ @use '/node_modules/vite-plugin-vue-devtools/src/overlay/devtools-overlay.css';
[0]     │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[0]     ╵
[0]     src\assets\base.scss 2:1           @use
[0]     src\content-script\index.scss 1:1  root stylesheet
```

Correct me if I am testing this in a wrong manner please, or let me know how to fix this, thanks in advance

**Comment by mubaidr** [comment]
Did you do `pnpm i` before the dev?

**Comment by maenkhouseh** [comment]
of course

**Comment by maenkhouseh** [comment]
in base.scss
using this:
`@use '../../node_modules/vite-plugin-vue-devtools/src/overlay/devtools-overlay.css';
`

fixes the issue

daisyui works now in the shadow-dom version?

**Comment by mubaidr** [comment]
What os are you on?

**Comment by maenkhouseh** [comment]
currently windows

**Comment by mubaidr** [comment]
Just tested on linux, no errors during dev or build. 

```bash
pnpm dev:chrome

> vite-vue3-chrome-extension-v3@0.0.1 dev:chrome /projects/vite-vue3-chrome-extension-v3
> vite -c vite.chrome.config.ts

Re-optimizing dependencies because lockfile has changed

  VITE v6.0.3  ready in 4783 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: use --host to expose
  ➜  Vue DevTools: Open http://localhost:5173/__devtools__/ as a separate window
  ➜  Vue DevTools: Press Alt(⌥)+Shift(⇧)+D in App to toggle the Vue DevTools

  ➜  press h + enter to show help

```

Maybe we need to test on `windows` specifically?

**Comment by maenkhouseh** [comment]
Yes, once updated, it worked and I ll test build and let you know

**Comment by maenkhouseh** [comment]
but daisyui seems not working, any fix?

**Comment by maenkhouseh** [comment]
Tried many things to make it work but none worked, any suggestions?

**Comment by maenkhouseh** [comment]
Any news?

**Comment by maenkhouseh** [comment]
@mubaidr is there a fix for this?

**Comment by mubaidr** [comment]
Apologies, but I did not have time to work further on tis. I will do, when i have time though.

**Reply by maenkhouseh** [reply]
Many thanks and much appreciated

**Reply by mubaidr** [reply]
I will try to debug this issue tonight, but can you explain why `iframe` is not valid option for this? In the latest branch, I have updated styles import to `iframe`, so it won't influence the parent webpage and vice versa.

**Reply by mubaidr** [reply]
@maenkhouseh I am looking for your feedback here, before I try to debug issue with `shadow-dom` branch.

**Comment by mubaidr** [comment]
Ohh And i almost forgot to say thank you for sponsoring this feature! Really appreciate that! @maenkhouseh

**Reply by maenkhouseh** [reply]
You are most welcome, and your work much appreciated.

**Comment by mubaidr** [comment]
As you can see in attached screenshot, daisy ui webpage has different theme and our content script has different theme. 

![Screenshot_20241227_000344](https://github.com/user-attachments/assets/105f59e2-617f-4277-bdc1-6377755936de)
